### ISSUE TITLE
list: checkbox secondary action gives aria-label warning 

### ISSUE THREAD

**Issue by jamalx31** [angular/material#6152]
I get the warning:
ARIA: Attribute " aria-label ", required for accessibility, is missing on node

My element is:
`<md-checkbox aria-label="Follow" class="md-secondary" ng-init="..." ng-model=".." ng-click="..."></md-checkbox>`

inside md-list-item
When I check the html code of the page I see that the parent of the checkbox is a `<bottom>` with the aria-label

**Comment by DevVersion** [comment]
I'm trying to reproduce that.

http://codepen.io/DevVersion/pen/KVwRjv

Can you take a look? (I don't receive any warning)

**Comment by marcysutton** [comment]
Is it because of the `ng-init`? Is there something happening in that step that interferes with the label warning utility?

**Comment by jamalx31** [comment]
@DevVersion 
Try this http://codepen.io/anon/pen/KVpyXE

@marcysutton 
I think the problem is with the ng-click

**Comment by crisbeto** [comment]
This is a pretty specific set of circumstances, but it happens because the `md-list-item` [removes the `aria-label` and `ng-click`](https://github.com/angular/material/blob/master/src/components/list/list.js#L181), from children that happen to have the `md-secondary` class and [adds them to itself](https://github.com/angular/material/blob/master/src/components/list/list.js#L201). E.g. if you remove the `ng-click` or the `md-secondary`, the warning goes away. 

I'm not sure whther it would be a good idea to disable the expectation if the checkbox's parent an aria-label.

**Comment by marcysutton** [comment]
Labeling is all about what element takes focus. Because mdCheckbox is a custom element and not a native checkbox, we use `aria-label` to accomplish labeling for screen readers. The label should go right onto `md-checkbox`, since it is normally a focusable, interactive element–the label text can't cascade down from a parent element or bubble up from a child.

However, there are bigger problems with this control than the label. `md-list` is wrapping the checkbox control in a `<button>`, which is not right at all. Both of those are independent components with different roles and purposes–the button shouldn't wrap the checkbox, even if it has `tabindex="-1"` on it. Due to this, the label is ending up on a button, yet `md-checkbox` internally requires its own label to be exposed, hence the warning.

Fixing this issue depends on a change to `md-list` to not wrap the checkbox in a button. @ThomasBurleson should we file a new issue for that?

**Comment by crisbeto** [comment]
I'll put this back into the backlog, because it requires a restructure of the list component.

**Comment by Splaktar** [comment]
Reopening this as the PR that closed it was reverted in `1.1.2` and I can reproduce it in this [updated CodePen](https://codepen.io/Splaktar/pen/OJLrrME?editors=1011#0).

**Comment by Splaktar** [comment]
I've found that this is due to the demo including naked `text` rather than `<p>text</p>`.

But we also support `<span>text</span>` inside of an `md-list-item` and that is currently throwing these warnings and failing to apply an appropriate `aria-label`.
